homeassistant/components/tplink/camera.py

```diff
@@ -0,0 +1,150 @@
+    CameraEntityFeature,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.aiohttp_client import async_aiohttp_proxy_stream
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TPLinkConfigEntry, legacy_device_id
+from .coordinator import TPLinkDataUpdateCoordinator
+from .entity import CoordinatedTPLinkEntity, TPLinkModuleEntityDescription
+
+FORCE_REFRESH_INTERVAL = timedelta(minutes=3)
+MOTION_DETECTION_CAPABILITY = "motion_detection"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class TPLinkCameraEntityDescription(
+    CameraEntityDescription, TPLinkModuleEntityDescription
+):
+    """Base class for event entity description."""
+
+
+CAMERA_DESCRIPTIONS: tuple[TPLinkCameraEntityDescription, ...] = (
+    TPLinkCameraEntityDescription(
+        key="live",
+        translation_key="live",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: TPLinkConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up climate entities."""
+    data = config_entry.runtime_data
+    parent_coordinator = data.parent_coordinator
+    device = parent_coordinator.device
+    ffmpeg_manager = ffmpeg.get_ffmpeg_manager(hass)
+
+    async_add_entities(
+        TPLinkCameraEntity(
+            device,
+            parent_coordinator,
+            description,
+            camera_module=camera_module,
+            parent=None,
+            ffmpeg_manager=ffmpeg_manager,
+        )
+        for description in CAMERA_DESCRIPTIONS
+        if (camera_module := device.modules.get(Module.Camera))
+    )
+
+
+class TPLinkCameraEntity(CoordinatedTPLinkEntity, Camera):
+    """Representation of a TPLink thermostat."""
+
+    _attr_name = None
+    _attr_supported_features = CameraEntityFeature.STREAM | CameraEntityFeature.ON_OFF
+
+    def __init__(
+        self,
+        device: Device,
+        coordinator: TPLinkDataUpdateCoordinator,
+        description: TPLinkCameraEntityDescription,
+        *,
+        camera_module: CameraModule,
+        parent: Device | None = None,
+        ffmpeg_manager: ffmpeg.FFmpegManager,
+    ) -> None:
+        """Initialize a Ring Door Bell camera."""
+        self._description = description
+        super().__init__(device, coordinator, parent=parent)
+        Camera.__init__(self)
+        self._camera_module = camera_module
+        self._ffmpeg_manager = ffmpeg_manager
+
+        self._video_url: str | None = None
+        self._image: bytes | None = None
+
+    def _get_unique_id(self) -> str:
+        """Return unique ID for the entity."""
+        return f"{legacy_device_id(self._device)}-{self._description}"
+
+    @callback
+    def _async_update_attrs(self) -> None:
+        """Update the entity's attributes."""
+        self._attr_is_on = self._camera_module.is_on
+        self._video_url = self._camera_module.stream_rtsp_url()
```

Comment:
`stream_rtsp_url()` is empty for my camera as it is using the credentials hash for logging in -> no feed available.

Reply:
Will be resolved by https://github.com/python-kasa/python-kasa/pull/1293

Reply:
Just a side note unrelated to this PR, should we make this a property in python-kasa?

Reply:
Convert `stream_rtsp_url` to be a property? I'd say no because credentials and stream resolution can be passed in to this function.